### ISSUE TITLE
[2022/11/30] Gradient Descent: The Ultimate Optimizer

### ISSUE THREAD

**Issue by jonghwanhyeon** [zaemyung/Seeking-SOTA#70]
- 제목: [Gradient Descent: The Ultimate Optimizer](https://openreview.net/forum?id=-Qp-3L-5ZdI)
- 출판: NeurIPS 2022
- 저자: Kartik Chandra, Audrey Xie, Jonathan Ragan-Kelley, Erik Meijer
- 코드: https://github.com/kach/gradient-descent-the-ultimate-optimizer

> Working with any gradient-based machine learning algorithm involves the tedious task of tuning the optimizer’s hyperparameters, such as its step size. Recent work has shown how the step size can itself be optimized alongside the model parameters by manually deriving expressions for “hypergradients” ahead of time. We show how to automatically compute hypergradients with a simple and elegant modification to backpropagation. This allows us to easily apply the method to other optimizers and hyperparameters (e.g. momentum coefficients). We can even recursively apply the method to its own hyper-hyperparameters, and so on ad infinitum. As these towers of optimizers grow taller, they become less sensitive to the initial choice of hyperparameters. We present experiments validating this for MLPs, CNNs, and RNNs. Finally, we provide a simple PyTorch implementation of this algorithm (see people.csail.mit.edu/kach/gradient-descent-the-ultimate-optimizer).

**Comment by jonghwanhyeon** [comment]
[Gradient Descent - The Ultimate Optimizer .pdf](https://github.com/zaemyung/Seeking-SOTA/files/10123139/Gradient.Descent.-.The.Ultimate.Optimizer.pdf)

**Comment by zaemyung** [comment]
https://aihub.org/2022/11/30/neurips2022-outstanding-paper-gradient-descent-the-ultimate-optimizer/
